lib/semantics/check-do.cc

```diff
@@ -219,6 +214,46 @@ class DoConcurrentBodyEnforce {
   }
 
 private:
+  void CheckAllocateObjectIsCoarray(
+      const parser::AllocateObject &allocateObject, const char *stmtType) {
+    const Symbol *symbol{nullptr};
+    parser::CharBlock position;
+    std::visit(
+        common::visitors{
+            [&](const parser::Name &x) {
+              symbol = x.symbol;
+              position = x.source;
+            },
+            [&](const parser::StructureComponent &x) {
+              symbol = x.component.symbol;
+              position = x.component.source;
+            },
+        },
+        allocateObject.u);
+    if (IsCoarray(*symbol)) {
+      context_.Say(position,
+          "%s coarray not allowed in DO CONCURRENT"_err_en_US, stmtType);
+    }
+  }
+
+  void CheckAllocateContainsCoarray(const parser::AllocateStmt &allocateStmt) {
+    const std::list<parser::Allocation> &allocationList{
+        std::get<1>(allocateStmt.t)};
```

Comment:
I think it's better to use `std::get` with a type rather than a number. It's shorter and avoids the magic number:
`const auto &allocationList{std::get<std::list<parser::Allocation>>};`

There are some in the next function too.

Reply:
Thanks, Tim.  I made these changes.